### PR TITLE
Do not allow trailing comma in a variable declaration.

### LINKED ISSUE #1360: incorrectly allows trailing commas in variable declaration statements
esprima accepts these programs, but should not:

```js
var a,
```

```js
var a, b,
```

```js
var a = 0,
```

```js
var a, b = 0,
```

### PR DESCRIPTION
**Pull request by ariya**
Fixes #1360

### CODE REVIEW

File: esprima.js [hunk 1]
```diff
@@ -4059,15 +4059,18 @@
...
+
+        opt = { inFor: options.inFor };
+        list = [parseVariableDeclaration(opt)];
 
-        do {
-            list.push(parseVariableDeclaration({ inFor: options.inFor }));
+        while (match(',')) {
+            lex();
+            list.push(parseVariableDeclaration(opt));
             if (!match(',')) {
```
Comment (michaelficarra) [inline esprima.js]: This should no longer be needed, since the very next test is exactly the same.

Comment (ariya) [inline esprima.js]: Good point, I'll remove the check.

**Comment by michaelficarra** [comment]
LGTM.

**Comment by mikesherov** [comment]
:+1:

**Comment by ariya** [comment]
Note that I will also forward port this to the master branch.

**Comment by ariya** [comment]
Landed in 2.x (for some reason GitHub doesn't close this PR).

**Pull request closed without merging**